speech_recognition/__init__.py

```diff
@@ -1666,6 +1667,44 @@ def recognize_tensorflow(self, audio_data, tensor_graph='tensorflow-data/conv_ac
             for node_id in top_k:
                 human_string = self.tflabels[node_id]
                 return human_string
+
+    def recognize_whisper(self, audio_data, model="base", show_dict=False, load_options=None, language=None, translate=False, **transcribe_options):
+        """
+        Performs speech recognition on ``audio_data`` (an ``AudioData`` instance), using Whisper.
+
+        The recognition language is determined by ``language``, an uncapitalized full language name like "english" or "chinese". See the full language list at https://github.com/openai/whisper/blob/main/whisper/tokenizer.py
+
+        model can be any of tiny, base, small, medium, large, tiny.en, base.en, small.en, medium.en. See https://github.com/openai/whisper for more details.
+
+        If show_dict is true, returns the full dict response from Whisper, including the detected language. Otherwise returns only the transcription.
+
+        You can translate the result to english with Whisper by passing translate=True
+
+        Other values are passed directly to whisper. See https://github.com/openai/whisper/blob/main/whisper/transcribe.py for all options
+        """
+
+        assert isinstance(audio_data, AudioData), "Data must be audio data"
+        import whisper
+
+        if load_options or not hasattr(self, "whisper_model") or self.whisper_model.get(model) is None:
```

Comment:
memo: `or` is short-circuit.

https://docs.python.org/3/reference/expressions.html#boolean-operations
>The expression x or y first evaluates x; if x is true, its value is returned; otherwise, y is evaluated and the resulting value is returned.

- When you passed not empty `dict` as `load_options`, load model
- When `load_options` is `None` or `{}` and the instance does not have `whisper_model` attribute, then load model
- When `load_options` is `None` or `{}` and the instance have `whisper_model` attribute but the name `model` does not included, then load model